lib/phoenix_live_view/js.ex

```diff
@@ -173,6 +173,37 @@ defmodule Phoenix.LiveView.JS do
 
   You can also use `window.addEventListener` to listen to events pushed
   from the server. You can learn more in our [JS interoperability guide](js-interop.md).
+
+  ## Composing JS commands
+
+  All the functions in this module optionally accept an existing `%JS{}` struct as the first argument,
+  allowing you to chain multiple commands, like pushing an event to the server and optimistically hiding
+  a modal:
+
+  ```heex
+  <div id="modal" class="modal">
+    My Modal
+  </div>
+
+  <button phx-click={JS.push("modal-closed") |> JS.remove_class("show", to: "#modal", transition: "fade-out")}>
+    hide modal
+  </button>
+  ```
+
+  Note that the commands themselves are executed on the client in a the order they are composed
+  and the client does not wait for a confirmation before executing the next command. If you chain
+  `JS.push(...) |> JS.hide(...)`, since hide is a fully client-side command, it hides immediately
+  after pushing the event, not waiting for the server to respond. If you chain multiple commands that
+  need to interact with the server, those are also guaranteed to be executed in the order they are composed,
+  since LiveView itself can also only handle one event at a time. Therefore, if you do something like
+
+  ```elixir
+  JS.push("my-event") |> JS.patch("/my-path?foo=bar")
+  ```
+
+  it is guaranteed that the event will be pushed first and the patch will only be handled after
+  the first event was processed by the LiveView.
```

Comment:
Maybe "… was handled by the LiveView"? Processed feels imprecise enough, where it could also mean (received, but only queued for handling). `handle` at least by name matches `handle_event`.